hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/ChangedReadersObserver.java

```diff
@@ -40,5 +40,16 @@
    * @param memStoreScanners scanner of current memstore
    * @throws IOException e
    */
-  void updateReaders(List<HStoreFile> sfs, List<KeyValueScanner> memStoreScanners) throws IOException;
+  void updateReaders(List<HStoreFile> sfs, List<KeyValueScanner> memStoreScanners)
+    throws IOException;
+
+  /**
+   * Notify observers to close all scanners and re-create them from given list of store files -
+   * should be non-compacted store files
+   *
+   * @param storeFiles non compacted store files to re-create scanners from
+   * @throws IOException if something goes wrong while dealing with store scanners
+   */
+  void updateReaders(List<HStoreFile> storeFiles) throws IOException;
```

Comment:
Why not pass the compacted away files and new files alone here?  In the above updateReaders method, the new flushed file(s) are only being passed

Reply:
This should be list of current HFiles that store has and not compacted away files. Also, in case of updateReaders during flush, memStoreScanners are passed that are in use but for compaction, we don't know which specific KeyValueScanners are on compacted sore files or on new files. In face, the impl is supposed to figure that all, hence list of KVScanners are not required.

Reply:
Just the same API is enough for us ? We can pass an empty list for memstoreScanners ?

Reply:
Same API assumes that we need to get rid of last memscanner but new one doesn't need to touch memscanners. Also, new API needs to clear currentScanners map